cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/common/DataprocUtils.java

```diff
@@ -70,6 +73,35 @@ public static String getBucketName(String bucket) {
     return bucket;
   }
 
+  /**
+   * Utilty class to parse the keyvalue string from UI Widget and return back HashMap.
+   * String is of format  <key><keyValueDelimiter><value><delimiter><key><keyValueDelimiter><value>
+   * eg:  networktag1=out2internet;networktag2=priority
+   * The return from the method is a map with key value pairs of (networktag1 out2internet) and (networktag2 priority)
+   *
+   * @param configValue String to be parsed into key values format
+   * @param delimiter Delimiter used for keyvalue pairs
+   * @param keyValueDelimiter Delimiter between key and value.
+   * @return Map of Key value pairs parsed from input configValue using the delimiters.
+   */
+  public static Map<String, String> parseKeyValueConfig(String configValue, String delimiter,
+                                                        String keyValueDelimiter) throws IllegalArgumentException {
+    Map<String, String> map = new HashMap<>();
+    if (configValue == null) {
```

Comment:
if `configValue` can be `null`, annotate it with `@Nullable` in the method parameter

Reply:
Done